Source/Scene/Scene.js

```diff
@@ -3645,6 +3664,85 @@ define([
         camera.right = right;
     }
 
+    function updateAsyncLoader(scene, asyncLoader) {
+        var context = scene._context;
+        var uniformState = context.uniformState;
+        var frameState = scene._frameState;
+
+        var view = scene._pickOffscreenView;
+        scene._view = view;
+
+        var ray = asyncLoader.ray;
+        var primitives = asyncLoader.primitives;
+
+        updateCameraFromRay(ray, view.camera);
+
+        updateFrameState(scene);
+        frameState.passes.offscreen = true;
+        frameState.passes.async = true;
+
+        uniformState.update(frameState);
+
+        var commandList = frameState.commandList;
+        var commandsLength = commandList.length;
+
+        var ready = true;
+        var primitivesLength = primitives.length;
+        for (var i = 0; i < primitivesLength; ++i) {
+            var primitive = primitives[i];
+            if (scene.primitives.contains(primitive) && primitive.show) {
```

Comment:
Minor, but isn't `primitive.show && scene.primitives.contains(primitive)` faster to avoid `contains` when show is false?